js/models/SMTPSettings.js

```diff
@@ -1,9 +1,50 @@
 import BaseModel from './BaseModel';
+import app from '../app';
+import is from 'is_js';
 
 export default class extends BaseModel {
   defaults() {
     return {
-      Notifications: true,
+      notifications: false,
+      serverAddress: '',
+      username: '',
+      password: '',
+      senderEmail: '',
+      recipientEmail: '',
     };
   }
+
+  validate(attrs) {
+    const errObj = {};
+    const addError = (fieldName, error) => {
+      errObj[fieldName] = errObj[fieldName] || [];
+      errObj[fieldName].push(error);
+    };
+
+    if (attrs.notifications) {
+      if (is.not.url(attrs.serverAddress.trim())) {
+        addError('serverAddress', app.polyglot.t('smtpModelErrors.serverAddress'));
+      }
+      if (is.empty(attrs.username.trim())) {
+        addError('username', app.polyglot.t('smtpModelErrors.username.caseEmpty'));
+      } else if (is.not.alphaNumeric(attrs.username.trim())) {
+        addError('username', app.polyglot.t('smtpModelErrors.username.caseInvalid'));
+      }
+      if (is.empty(attrs.password.trim())) {
+        addError('password', app.polyglot.t('smtpModelErrors.password'));
+      }
+      if (is.not.email(attrs.senderEmail.trim())) {
+        addError('senderEmail', app.polyglot.t('smtpModelErrors.senderEmail'));
+      }
+      if (is.not.email(attrs.recipientEmail.trim())) {
+        addError('recipientEmail', app.polyglot.t('smtpModelErrors.recipientEmail'));
+      }
+    }
```

Comment:
Something is still broken in validation, when I try deleting an address, this error is thrown:

```
Uncaught Error: Client side validation failed: {"smtpSettings.serverAddress":["Please provide a valid URL."],"smtpSettings.username":["Please provide a username."],"smtpSettings.password":["Please provide a valid password."],"smtpSettings.senderEmail":["Please provide a valid email."],"smtpSettings.recipientEmail":["Please provide a valid email."]}
```

Saving the addresses shouldn't result in the SMTP settings model validation failing.

**Edit: it looks like this is because I had notifications set to true, with no values for the other fields.** While that shouldn't happen in the wild once this is merged, we should handle that situation instead of just having the client break. 

For the situations where there are multiple models all being saved to one API, how do we want to handle cross-tab errors? If saving an address throws an error in another tab, we should alert the user to that so they can fix it.